/**!
 * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

option java_package = "com.snowflake.apps.streamlit";


import "streamlit/proto/AppPage.proto";

message Navigation {
  repeated string sections = 1;
  repeated AppPage app_pages = 2;
  Position position = 3;

  // The script hash for the page identified by st.navigation
  string page_script_hash = 4;

  // Position of the Navigation
  enum Position {
    // Plain, fixed width text.
    SIDEBAR = 0;

    // Shows an error message.
    HIDDEN = 1;
  }
}
